components/DiscordBar.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from 'react'
+import styles from '../scss/discordBar.module.scss'
+type Member = {
+  id: string
+  username: string
+  status: string
+  avatar_url: string
+}
+interface discordJSON {
+  members?: Member[]
+}
+const ONE_MINUTE = 60 * 1000
+const Member: React.FC<{ avatar: string }> = ({ avatar }) => {
+  return (
+    <div className="d-inline-block">
+      <img src={avatar} className={`${styles['member-avatar']} mx-1`} />
+    </div>
+  )
+}
+const DiscordBar: React.FC = () => {
+  interface State extends discordJSON {
+    error?: string
+  }
+  const [data, setData] = useState<State>({
+    members: []
+  })
+  const getData = async () => {
+    try {
+      const data = await fetch(
+        'https://discord.com/api/guilds/828783458469675019/widget.json'
+      )
+      const json: discordJSON = await data.json()
+      setData(json)
```

Comment:
Check if `members` exists in the data you fetched to simplify the checks in the rendering code. You can chain the `.then()` part after the fetch, the `await` command will work the same. You can then define `members` as a non null value.

P.S.: I didn't test this code, let me know if it doesn't work.

```suggestion
      const json = await fetch(
        'https://discord.com/api/guilds/828783458469675019/widget.json'
      ).then(r => r.json())
      if ('members' in json) setData(json as discordJson)
      else throw new Error() // fallback to the catch block
```

Reply:
I added new error case for no members data. Error during fetching will result in `Error fetching data...` message while successful response without members data is generic `Something went wrong ;(`